web/src/app/schedules/ScheduleRuleCreateDialog.tsx

```diff
@@ -46,51 +54,44 @@ export default function ScheduleRuleCreateDialog(props) {
     ],
   })
 
-  const [mutate, mutationStatus] = useMutation(mutation, {
-    onCompleted: onClose,
-    variables: {
-      input: {
-        target: {
-          type: targetType,
-          id: value.targetID,
-        },
-        scheduleID,
-
-        rules: value.rules.map((r) => ({
-          ...r,
-          start: isoToGQLClockTime(r.start, data.schedule.timeZone),
-          end: isoToGQLClockTime(r.end, data.schedule.timeZone),
-        })),
-      },
-    },
-  })
+  const [{ error }, mutate] = useMutation(mutation)
 
   return (
     <FormDialog
       onClose={onClose}
       title={`Add ${startCase(targetType)} to Schedule`}
-      errors={nonFieldErrors(mutationStatus.error)}
+      errors={nonFieldErrors(error)}
       maxWidth='md'
-      loading={(!data && queryStatus.loading) || mutationStatus.loading}
-      onSubmit={() => {
-        mutate()
-      }}
+      loading={!data && fetching}
+      onSubmit={() =>
+        mutate(
+          {
+            input: {
+              target: {
+                type: targetType,
+                id: value.targetID,
+              },
+              scheduleID,
+
+              rules: value.rules.map((r) => ({
+                ...r,
+                start: isoToGQLClockTime(r.start, data.schedule.timeZone),
+                end: isoToGQLClockTime(r.end, data.schedule.timeZone),
+              })),
+            },
+          },
+          { additionalTypenames: ['Schedule'] },
+        ).then(() => onClose())
```

Comment:
we'll want to check if there's any errors first here:
```suggestion
        ).then((result) => {
          if (!result.error) onClose()
        })
```